resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/tpr/TPRCrudCallsImpl.scala

```diff
@@ -0,0 +1,320 @@
+      .url(url))
+
+    logDebug(s"Create SparkJobResource Request: $request")
+    val response = httpClient.newCall(request).execute()
+    completeRequestWithExceptionIfNotSuccessful(
+      "post",
+      response,
+      Option(Seq(name, response.toString, compact(render(payload)))))
+
+    response.body().close()
+    logDebug(s"Successfully posted resource $name: " +
+      s"${pretty(render(parse(write(resourceObject))))}")
+  }
+
+  override def deleteJobObject(tprObjectName: String): Unit = {
+    val requestSegments = Seq(
+      "apis", TPR_API_GROUP, TPR_API_VERSION, "namespaces", namespace, "sparkjobs", tprObjectName)
+    val url = generateHttpUrl(requestSegments)
+
+    val request = completeRequest(new Request.Builder()
+      .delete()
+      .url(url))
+
+    logDebug(s"Delete Request: $request")
+    val response = httpClient.newCall(request).execute()
+    completeRequestWithExceptionIfNotSuccessful(
+      "delete",
+      response,
+      Option(Seq(tprObjectName, response.message(), request.toString)))
+
+    response.body().close()
+    logInfo(s"Successfully deleted resource $tprObjectName")
+  }
+
+  override def getJobObject(name: String): SparkJobState = {
+    val requestSegments = Seq(
+      "apis", TPR_API_GROUP, TPR_API_VERSION, "namespaces", namespace, "sparkjobs", name)
+    val url = generateHttpUrl(requestSegments)
+
+    val request = completeRequest(new Request.Builder()
+      .get()
+      .url(url))
+
+    logDebug(s"Get Request: $request")
+    val response = httpClient.newCall(request).execute()
+    completeRequestWithExceptionIfNotSuccessful(
+      "get",
+      response,
+      Option(Seq(name, response.message()))
+    )
+
+    logInfo(s"Successfully retrieved resource $name")
+    read[SparkJobState](response.body().string())
+  }
+
+  override def updateJobObject(name: String, value: String, fieldPath: String): Unit = {
+    val payload = List(
+      ("op" -> "replace") ~ ("path" -> fieldPath) ~ ("value" -> value))
+    val requestBody =
+      RequestBody.create(
+        MediaType.parse("application/json-patch+json"),
+        compact(render(payload)))
+
+    val requestSegments = Seq(
+      "apis", TPR_API_GROUP, TPR_API_VERSION, "namespaces", namespace, "sparkjobs", name)
+    val url = generateHttpUrl(requestSegments)
+
+    val request = completeRequest(new Request.Builder()
+      .patch(requestBody)
+      .url(url))
+
+    logDebug(s"Update Request: $request")
+    val response = httpClient.newCall(request).execute()
+    completeRequestWithExceptionIfNotSuccessful(
+      "patch",
+      response,
+      Option(Seq(name, response.message(), compact(render(payload))))
+    )
+
+    response.body().close()
+    logDebug(s"Successfully patched resource $name.")
+  }
+
+/**
+ * This method has an helper method that blocks to watch the object.
+ * The future is completed on a Delete event or source exhaustion.
+ * This method also relies on the assumption of one sparkjob per namespace
```

Comment:
Since each spark job deployed is attached to a SparkJob resource, I believe this assumption holds as long as, we still run one sparkjob per namespace.